kombu/transport/SQS.py

```diff
@@ -899,6 +943,54 @@ def sqs_base64_encoding(self):
     def fetch_message_attributes(self):
         return self.transport_options.get('fetch_message_attributes')
 
+    @property
+    def get_message_attributes(self) -> dict[str, Any]:
+        """Get the message attributes to be fetched from SQS.
+
+        Ensures 'ApproximateReceiveCount' is included in system attributes if list is provided.
+        - The number of retries is managed by SQS /
+            (specifically by the ``ApproximateReceiveCount`` message attribute)
+        - See: class QoS(virtual.QoS):
+            (method) def extract_task_name_and_number_of_retries
+
+        :return: A dictionary with SQS message attribute fetch config.
+        """
+        fetch = self.fetch_message_attributes
```

Comment:
can you please elaborate more on fetch_message_attributes?

Reply:
This comes from the existing `@cached_property`, which grabs the value from the `transport_options`. I use that value and parse out if it is a `list`, as was expected before, or the new `dict` that includes both types of message attributes.

```    
@cached_property
def fetch_message_attributes(self):
        return self.transport_options.get('fetch_message_attributes', None)
```